api/src/main/java/com/redhat/ipaas/api/v1/rest/DataManager.java

```diff
@@ -182,74 +194,106 @@ public String generatePK(Map<String, WithId> entityMap) {
-        Map<String, WithId> entityMap = cache.computeIfAbsent(model, k -> new HashMap<>());
+        String kind = entity.kind();
+        Cache<String, WithId> cache = caches.getCache(kind);
         Optional<String> id = entity.getId();
         String idVal;
         if (!id.isPresent()) {
-            idVal = generatePK(entityMap);
+            idVal = generatePK(cache);
             entity = (T) entity.withId(idVal);
         } else {
             idVal = id.get();
-            if (entityMap.keySet().contains(idVal)) {
+            if (cache.keySet().contains(idVal)) {
                 throw new EntityExistsException("There already exists a "
-                    + model + " with id " + idVal);
+                    + kind + " with id " + idVal);
             }
         }
-        entityMap.put(idVal, entity);
-        //TODO interact with the back-end system
+
+        T finalEntity = entity;
+        cache.put(idVal, (T) doWithDataAccessObject(kind, d -> d.create(finalEntity)));
         return entity;
     }
 
     public void update(WithId entity) {
-        String model = entity.kind();
-        Map<String, WithId> entityMap = cache.get(model);
+        String kind = entity.kind();
+        Map<String, WithId> cache = caches.getCache(kind);
+
         Optional<String> id = entity.getId();
         if (!id.isPresent()) {
             throw new EntityNotFoundException("Setting the id on the entity is required for updates");
         }
+
         String idVal = id.get();
-        if (entityMap == null || !entityMap.containsKey(idVal)) {
-            throw new EntityNotFoundException("Can not find " + model + " with id " + idVal);
+        if (!cache.containsKey(idVal)) {
+            throw new EntityNotFoundException("Can not find " + kind + " with id " + idVal);
         }
+
+        doWithDataAccessObject(kind, d -> d.update(entity));
+        cache.put(idVal, entity);
         //TODO 1. properly merge the data ? + add data validation in the REST Resource
-        //TODO 2. interact with the back-end system
-        entityMap.put(idVal, entity);
     }
 
-    public void delete(String model, String id) {
-        Map<String, WithId> entityMap = cache.get(model);
+
+    public boolean delete(String kind, String id) {
+        Map<String, WithId> cache = caches.getCache(kind);
         if (id == null || id.equals(""))
             throw new EntityNotFoundException("Setting the id on the entity is required for updates");
-        if (entityMap == null || !entityMap.containsKey(id))
-            throw new EntityNotFoundException("Can not find " + model + " with id " + id);
-        //TODO interact with the back-end system
-        entityMap.remove(id);
+        if (!cache.containsKey(id))
+            throw new EntityNotFoundException("Can not find " + kind + " with id " + id);
+
+        WithId entity = cache.get(id);
+        if (entity != null && doWithDataAccessObject(kind, d -> d.delete(entity))) {
+            cache.remove(id);
```

Comment:
I like this idea, we should definitely try it out. Still, I see no harm in removing it here.